cupy/core/core.pyx

```diff
@@ -1011,14 +1012,19 @@ cdef class ndarray:
         cdef Py_ssize_t i, j, offset, ndim, n_newaxes, n_ellipses, ellipsis
         cdef Py_ssize_t ellipsis_sizem, s_start, s_stop, s_step, dim, ind
         cdef vector.vector[Py_ssize_t] shape, strides
-        if not isinstance(slices, tuple):
+        if isinstance(slices, ndarray):
+            pass
+        elif isinstance(slices, numpy.ndarray):
+            slices = cupy.asarray(slices)
+        elif not isinstance(slices, tuple):
             slices = [slices]
         else:
             slices = list(slices)
 
-        for s in slices:
-            if isinstance(s, ndarray):
-                raise ValueError('Advanced indexing is not supported')
+        # boolean array indexing
+        if isinstance(slices, ndarray) and slices.dtype == bool:
+            assert internal.vector_equal(self._shape, slices._shape)
+            return _boolean_array_indexing(self, slices)
```

Comment:
As advanced indexing has not been fully supported yet, we still need to raise `ValueError` for unsupported slice types.